server/service/schedule/schedule.go

```diff
@@ -199,6 +203,27 @@ func (s *Schedule) Start() {
 						ctxerr.Handle(s.ctx, err)
 					}
 				}
+
+				totalRunningTime := time.Since(newStart)
+				schedInterval = s.getSchedInterval()
+
+				if totalRunningTime > schedInterval {
+					level.Info(s.logger).Log("cron", s.name, "msg", "total runtime (%d) exceeded schedule interval (%d)")
+				}
+
+				// set new wait to the remaining duration after dividing the total running time by the schedule interval
+				newWait = schedInterval - (totalRunningTime % schedInterval)
+				setWaitTimes(newStart, newWait)
+
+				select {
+				case <-schedTicker.C:
+					// pull from ticker channel in case another tick arrived during this run
+				default:
+					// ok
+				}
+				schedTicker.Reset(newWait)
+
+				cancelHold()
```

Comment:
We always cancel the hold. Lines 207-224 are just re-syncing the ticker in case the run lasts longer than the scheduled interval. But in either case, we cancel the hold before when loop around to wait on the next tick.

Reply:
But don't we want to cancel the holder _before_ we wait the rest of the time?